ProcessMaker/Repositories/CaseSyncRepository.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace ProcessMaker\Repositories;
+
+use Illuminate\Support\Collection;
+use ProcessMaker\Models\CaseParticipated;
+use ProcessMaker\Models\CaseStarted;
+use ProcessMaker\Models\ProcessRequest;
+
+class CaseSyncRepository
+{
+    /**
+     * Sync the cases started.
+     *
+     * @param array $requestIds
+     * @return array
+     */
+    public static function syncCases(array $requestIds): array
+    {
+        $requests = ProcessRequest::with([
+            'tokens', 'process', 'childRequests.process', 'childRequests.tokens', 'participants',
+        ])
+        ->whereIn('id', $requestIds)
+        ->get();
+
+        $successes = [];
+        $errors = [];
+
+        foreach ($requests as $instance) {
+            try {
+                if (!is_null($instance->parent_request_id)) {
+                    continue;
+                }
+
+                $csProcesses = CaseUtils::storeProcesses($instance, collect());
+                $csRequests = CaseUtils::storeRequests($instance, collect());
+                $csRequestTokens = collect();
+                $csTasks = collect();
+                $participants = $instance->participants->map->only('id', 'fullName', 'title', 'avatar');
+                $status = $instance->status === 'ACTIVE' ? 'IN_PROGRESS' : $instance->status;
```

Comment:
Use constants for the status strings. Create them in the right place if does not exists. E.g.

```
ProcessRequest::STATUS_ACTIVE = 'ACTIVE';
Cases::STATUS_IN_PROGRESS = 'IN_PROGRESS';
```

Reply:
It was fixed